h/static/scripts/vendor/jquery.scrollintoview.js

```diff
@@ -109,29 +109,37 @@
 
 				var animOptions = {};
 
+				var yDir = "none";
```

Comment:
> Let's simply not do this. This is all inane.

What specifically is inane?

> Even with all the work you just put into this,

(75 lines added, 60 lines removed ... not exactly a full reimplementation)

> the padding does not consider which element needs to scroll to produce that effect.

In theory, yes, but have you ever seen an example where it produces the wrong behavior?

> I'm still in favor of killing the bucket bar or, at least, the scrolling feature.

Even if we would kill the bucket bar (which I don't think we should do, without offering a better replacement), we would still have the scrolling feature, because it's also triggered by the user clicking on annotation cards in the sidebar. So killing the sidebar wouldn’t eliminate the need for this.

On the other hand, we could eliminate the padding feature, but it's nice, and people (both devs and users) gave positive feedback for it, and negative feedback for the lack of it. (https://github.com/hypothesis/h/issues/1776) So I would prefer to keep it.

> I still believe the feature does not serve any use case that so much needs such a feature that it justifies it's UI weight.

What UI weight?

> The scrolling has been a constant source of bugs

I don't remember all those bugs. At all.

>  and we've never fully implemented it to deal with all the cases that present.

Which cases are not dealt with?

> I say remove all this scrolling nonsense.

I say it's all perfectly fine.

Comparing some (guesstimated) numbers, I think it had less then 5% of the errors we had around our authentication, authorization and permission systems. Should we just remove all that, too? For what purpose?